map/src/test/java/openlr/map/simplemockdb/OpenLRMapDatabaseAdaptor.java

```diff
@@ -0,0 +1,137 @@
+package openlr.map.simplemockdb;
+
+import com.vividsolutions.jts.geom.Envelope;
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.LineString;
+import generated.SimpleMockedMapDatabase;
+import openlr.map.GeoCoordinates;
+import openlr.map.Line;
+import openlr.map.MapDatabase;
+import openlr.map.Node;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Unmarshaller;
+import java.awt.geom.Rectangle2D;
+import java.io.InputStream;
+import java.util.*;
+
+public class OpenLRMapDatabaseAdaptor implements MapDatabase {
+    private Map<Long, SimpleMockedNode> nodes = new HashMap<>();
+    private Map<Long, SimpleMockedLine> lines = new HashMap<>();
+    private SimpleMockedMapDatabase simpleMockedMapDatabase;
+
+    private OpenLRMapDatabaseAdaptor(SimpleMockedMapDatabase simpleMockedMapDatabase) {
+        this.simpleMockedMapDatabase = simpleMockedMapDatabase;
+    }
+
+    private static SimpleMockedMapDatabase parseXmlToSimpleMockedDatabase(InputStream xmlMap) throws SimpleMockedException {
+        try {
+            JAXBContext jaxbContext = JAXBContext.newInstance(SimpleMockedMapDatabase.class);
+
+            Unmarshaller unmarshaller = jaxbContext.createUnmarshaller();
+
+            return (SimpleMockedMapDatabase) unmarshaller.unmarshal(xmlMap);
+        } catch (JAXBException e) {
+            throw new SimpleMockedException(e.getMessage());
+        }
+    }
+
+
+    private void createNodes() {
+        this.simpleMockedMapDatabase.getNode().stream().forEach(node -> {
+            nodes.put(node.getId().longValue(), SimpleMockedNode.from(node));
+        });
+    }
+
+    private void createLines() {
+        simpleMockedMapDatabase.getLine().forEach(
+                line -> {
+                    Node startNode = this.nodes.get(line.getStart().longValue());
+                    Node endNode = this.nodes.get(line.getEnd().longValue());
+                    SimpleMockedLine simpleMockedLine = SimpleMockedLine.from(line, startNode, endNode);
+                    this.lines.put(simpleMockedLine.getID(), simpleMockedLine);
+                }
+        );
+    }
+
+    private void connectLines() {
+        this.nodes.forEach(
+                (id, node) -> {
+                    node.setConnections(lines.values());
+                }
+        );
+    }
+
```

Comment:
The initialisation logic is spread between static methods and private member methods.  I think it would be better to just build the nodes and lines in the static method and pass them to the constructor.  Once you have built the nodes and lines, you don't need the `SimpleMockedMapDatabase` anymore.  I will create a branch to show you what I mean.

Reply:
cc0fcab1679ad3d5ae77d3fb09c44fc5646d400c

Reply:
cc0fcab1679ad3d5ae77d3fb09c44fc5646d400c